apps/selvbetjening-ui/src/components/Navigation.tsx

```diff
@@ -3,20 +3,44 @@ import {Button} from "@navikt/ds-react";
 import styled, {css} from "styled-components";
 import FormGroup from "~common/FormGroup";
 import React from "react";
+import {useFormContext} from "react-hook-form";
 
 
 interface NavButtonProps {
-    label?: string
-    navigateTo?: string
+    text: string
+    navigateTo: string
+    onSubmit?: (form: any) => Promise<void>
 }
 
 export default function Navigation({left, right}: { left?: NavButtonProps, right?: NavButtonProps }) {
     const navigate = useNavigate()
+    const {handleSubmit, setError, clearErrors} = useFormContext()
+
+    const onClickWrapper = (button: NavButtonProps) => {
+        const {navigateTo, onSubmit} = button
+        if (onSubmit) {
+            const submitOgNaviger = async (data: any) => onSubmit(data).then(() => {
+                navigate(navigateTo)
+                clearErrors("root.serverError")
+            }).catch(() => {
+                setError("root.serverError", {message: "Noe gikk galt!"})
+            })
+            handleSubmit(submitOgNaviger)()
+        } else {
+            clearErrors("root.serverError")
+            navigate(navigateTo)
+        }
+    }
+
     return (
         <NavFooter>
             <NavRow>
-                {left && <Button onClick={() => left.navigateTo ? navigate(left.navigateTo) : ''}>{left.label}</Button>}
-                {right && <Button onClick={() => right.navigateTo ? navigate(right.navigateTo) : ''}>{right.label}</Button>}
+                {left && <Button onClick={() => {
+                    onClickWrapper(left)
+                }}>{left.text}</Button>}
+                {right && <Button onClick={() => {
+                    onClickWrapper(right)
+                }}>{right.text}</Button>}
```

Comment:
Hva er det egentlig som løses her?

Reply:
At man submiter formet med navigasjonsknappen istedenfor å ha en egen knapp for det
![Uploading image.png…]()

Reply:
IMO så synes jeg et sånt "magisk" skille mellom knappene som styres av skjemaet og selve skjemaet er litt uryddig. Kan gi deg et eksempelt når jeg er ferdig i møte